libs/nx-mcp/nx-mcp-server/src/lib/tools/nx-tasks.ts

```diff
@@ -0,0 +1,148 @@
+  server: McpServer,
+  logger: Logger,
+  telemetry?: NxConsoleTelemetryLogger,
+) {
+  server.tool(
+    NX_CURRENT_RUNNING_TASKS_DETAILS,
+    `Returns a list of running commands (also called tasks) from currently running Nx CLI processes. This will include the process ID of the Nx CLI processes with task IDs and their status.
+    There will be scenarios where the current process is not running anymore (as denoted by Stopped).
+    Use this tool if users ask for information about recently run tests, builds or other commands.
+    Use this tool for assisting with debugging and getting details about the current running tasks.
+
+    Use ${NX_CURRENT_RUNNING_TASK_OUTPUT} to get the terminal output for specific tasks.
+    `,
+    nxCurrentlyRunningTasksDetails(telemetry),
+  );
+  server.tool(
+    NX_CURRENT_RUNNING_TASK_OUTPUT,
+    `Returns the terminal output for a specific task from currently running Nx CLI processes`,
+    NxCurrentlyRunningTaskOutputSchema.shape,
+    nxCurrentlyRunningTaskOutput(telemetry),
+  );
+}
+
+const nxCurrentlyRunningTasksDetails =
+  (telemetry: NxConsoleTelemetryLogger | undefined) =>
+  async (): Promise<CallToolResult> => {
+    telemetry?.logUsage('ai.tool-call', {
+      tool: 'nx-currently_running_tasks_details',
+    });
+
+    const running_tasks = getRunningTasks();
+
+    const content: CallToolResult['content'] = [];
+    for (const task of running_tasks) {
+      content.push({
+        type: 'text',
+        text: `processId: ${task.processId} (${task.status})`,
+      });
+      for (const taskDetail of task.tasks) {
+        content.push({
+          type: 'text',
+          text: `-- taskId: ${taskDetail.name} (${taskDetail.status}) ${
+            taskDetail.continuous ? '(continuous)' : ''
+          }`,
+        });
+      }
+    }
+
+    return {
+      content,
+    };
+  };
+
+const NxCurrentlyRunningTaskOutputSchema = z.object({
+  processId: z
+    .number()
+    .optional()
+    .describe('The process ID of the Nx CLI process'),
+  taskId: z.string().describe('The task ID of the task to get the output for'),
+});
+
+type NxCurrentlyRunningTaskOutputType = z.infer<
+  typeof NxCurrentlyRunningTaskOutputSchema
+>;
+const nxCurrentlyRunningTaskOutput =
+  (telemetry: NxConsoleTelemetryLogger | undefined) =>
+  async ({
+    processId,
+    taskId,
+  }: NxCurrentlyRunningTaskOutputType): Promise<CallToolResult> => {
+    telemetry?.logUsage('ai.tool-call', {
+      tool: 'nx-currently_running_task_output',
+    });
+
+    const runningTasks = getRunningTasks();
+    const content: CallToolResult['content'] = [];
+
+    // If processId is specified, look for task within that specific process
+    if (processId) {
+      const process = runningTasks.find((task) => task.processId === processId);
+
+      if (!process) {
+        content.push({
+          type: 'text',
+          text: `No task found with process ID ${processId}`,
+        });
+        return { content };
```

Comment:
maybe we should specify `isError` to mark clearly that this is not what should happen